src/parser/helper/TreeParser.ts

```diff
@@ -13,9 +13,27 @@ export class TreeParser {
         }
 
         const parser = new Parser();
-        parser.setLanguage(grammars.get(parseFile.language));
+        parser.setLanguage(fileExtensionToGrammar.get(parseFile.fileExtension));
 
-        const sourceCode = fs.readFileSync(parseFile.filePath).toString();
+        const sourceCode = fs.readFileSync(parseFile.filePath, { encoding: "utf8" });
```

Comment:
Can this be changed to `fs.promises.readFile()` as well?

Reply:
`static async getParseTreeAsync` does already do this. This synchronous variant is used by the dependency analysis rather than the base file metrics. As mentioned in the opening of this PR, I restricted the scope of this PR on improving the calculation of the base file metrics. Improving the dependency analysis should be also done, of course.

Reply:
I have added #37 so that we do not forget this.